### PR TITLE
NEW: Implement converter for Velociprobe data

### PR DESCRIPTION
**Pull request by carterbox**
Implemented a converter for[ the data Junjing provided to us.](https://drive.google.com/drive/folders/1jIBwDuBcYVFkWvKJwLuONPYMo7ja5_Jz)

I added the ability to provide virtual datasets as input in order to accommodate for how Velociprobe data is stored as multiple files.

I reorganized the file structure to make imports more pythonic.

### CODE REVIEW

**Comment by JulReinhardt** [comment]
@carterbox today @prjemian  and I came to the conclusion that the NXpositioner should remain part of our definition as a simple way to access the positioning values for ptychography in case the full geometry was not (properly) defined in the NXtransformations and more importantly if we someone wants to define more than just raw positioner values but also target_values or anything else coming from e.g. interferometers.
 In case we do have both (full NXtransformation and NXpositioner), we simply link them as you already did in an earlier version of your APS converter example.

**Comment by carterbox** [comment]
OK. That's fine. I'll revert my example tomorrow.

**Review by JulReinhardt — CHANGES_REQUESTED** [review]

File: creator/velociprobe.py
Comment (carterbox) [reply creator/velociprobe.py:20]: Naming these parameters all 'h5parent' seems less 'self-documenting' when compared with naming them after the specific parent group they should be placed in.

Comment (JulReinhardt) [reply creator/velociprobe.py]: Ah, I see you chose "entry" as argument name, since most of them will have the entry group as h5parent. What about groups that don't go under entry, e.g. detector - should that be called "instrument" instead? And what about the transformation group that either has sample or detector groups as parent?

Comment (prjemian) [reply creator/velociprobe.py]: In NeXus, the NXinstrument group is a child of the NXentry group, so this default is the most common.  Alternatives would include `entry="entry1"`, etc.)

Comment (JulReinhardt) [reply creator/velociprobe.py]: Sure, what do you suggest now for the naming of the argument variable (not the specific input)?

Comment (prjemian) [reply creator/velociprobe.py]: I'm ok with the `entry` keyword as a name for the `create_instrument_group()` method.

Comment (JulReinhardt) [reply creator/velociprobe.py]: Sure, and then for create_detector_group we use a different one? I mean it does not really matter it's just a name. I only chose because that is what it is and it's general enough to be the same for all creator methods.

Comment (carterbox) [reply creator/velociprobe.py]: I was proposing that we use "h5parent" when the hierarchy is unknown such as for transformation, positioner groups, but when the hierarchy is known we use the name of the parent ("entry", "instrument") for groups such as instrument, beam, detector, sample, axis.

Similarly (but independently) we can also enforce the correct hierarchy by checking that the NX_class attribute of the parent is acceptable.

File: test/velociprobe.py
Comment (prjemian) [inline test/velociprobe.py:1]: Where is the _testing_ done?  This generates a NeXus file with specific imports.

Comment (carterbox) [reply test/velociprobe.py:1]: There's no format validator yet, so this "test" just checks that the converter executes without error.

Comment (prjemian) [reply test/velociprobe.py:1]: Could you refactor to use pytest and assert that the code does not raise an exception?

Comment (prjemian) [reply test/velociprobe.py:1]: This could be the beginning of implementing deeper tests as desired.

File: nxptycho/converter/velociprobe.py
Comment (prjemian) [inline nxptycho/converter/velociprobe.py:71]: The units are attributes.  The creator code checks that the supplied units can actually be converted into the expected units.  If not, it should report the error.  If no error, it (IIRC) writes the units as supplied.

File: nxptycho/converter/velociprobe.py
Comment (prjemian) [inline nxptycho/converter/velociprobe.py:109]: I believe this is correct.  Needs to be verified.

**Review by prjemian — CHANGES_REQUESTED** [review]
Just the one change for using pytest

File: creator/velociprobe.py
Comment (JulReinhardt) [reply creator/velociprobe.py]: yeah, I see pro and con for both approaches- naming all parent args _h5parent_ or only the ones where the parent is ambiguous and fixing others like instrument, beam etc. I'd prefer to name them all _h5parent_ and describe the default in doc, but happy to follow majority decision.

Yes let's check the NX_class attributes in addition no matter which convention we follow.

**Review by JulReinhardt — APPROVED** [review]
looks good to me, thanks for restructuring. We will have some dead folders/files after the merge that we can eliminate later.
Happy to merge, once Pete's happy.

File: test/velociprobe.py
Comment (carterbox) [reply test/velociprobe.py:1]: Converted examples to testing framework. Internet says that the default assumption is "no exception", so I don't need to "assert no exception".

File: nxptycho/converter/velociprobe.py
Comment (carterbox) [reply nxptycho/converter/velociprobe.py:109]: FWIW, the question I'm asking is whether additional attributes are added not whether adding additional attributes to an ExternalLink is forbidden (it is forbidden).

**Review by prjemian — APPROVED** [review]

**Pull request merged**
